app/src/main/java/com/github/orkest/Model/FireStoreDatabaseAPI.kt

```diff
@@ -56,6 +58,25 @@ class FireStoreDatabaseAPI {
 
         }
 
+    /**
+     * @param username of the user's follow lists to display
+     * @param isFollowersList is a boolean to indicate whether we want to fetch the followers list if true or the followings list if false
+     *
+     * @return the follow list indicated by the boolean isFollowersList of the user.
+     */
+    fun fetchFollowList(username: String, isFollowersList: Boolean): CompletableFuture<MutableList<String>>{
+        val future = CompletableFuture<MutableList<String>>()
+
+        getUserDocumentRef(username).get().addOnSuccessListener { document ->
+            if(document != null && document.exists()){
+                val user = document.toObject(User::class.java)
+                if (user != null) { if (isFollowersList) future.complete(user.followers) else future.complete(user.followings) }
+            }
+
+        }.addOnFailureListener { e -> Log.d(TAG, "Error fetching follow lists", e) }
```

Comment:
Maybe we should decide later of a common behavior to adopt when an error is raised but for now loggin is good

Reply:
Yes, you're right